test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightBatchWriterTests.cs

```diff
@@ -405,6 +409,112 @@ public async Task WriteBatchResponseWithChangesetAsync()
+
+                    // Copies operation's response body to the main batch response's stream
+                    // This is similar to how the DefaultODataBatchHandler in WebApi writes batch responses.  
+                    using (var customerResponseStream = CreateCustomerResponseStream(customerId: "1"))
+                    using (var stream = await customerOperationMessage.GetStreamAsync())
+                    {
+                        customerResponseStream.Seek(0L, SeekOrigin.Begin);
+                        await customerResponseStream.CopyToAsync(stream);
+                    }
+
+                    var orderOperationMessage = await jsonLightBatchWriter.CreateOperationResponseMessageAsync("2");
+                    orderOperationMessage.SetHeader("odata-version", "4.0");
+                    orderOperationMessage.SetHeader("Content-Type", "application/json");
+
+                    using (var orderOperationStream = CreateOrderResponseStream(orderId: "1"))
+                    using (var stream = await orderOperationMessage.GetStreamAsync())
+                    {
+                        orderOperationStream.Seek(0L, SeekOrigin.Begin);
+                        await orderOperationStream.CopyToAsync(stream);
+                    }
+
+                    await jsonLightBatchWriter.WriteEndBatchAsync();
+                },
+                writingRequest: false);
+
+            Assert.Equal("{\"responses\":[" +
+                "{\"id\":\"1\"," +
+                "\"status\":0," +
+                "\"headers\":{\"odata-version\":\"4.0\",\"content-type\":\"application/json\"}, " +
+                "\"body\" :{\"@odata.context\":\"http://tempuri.org/$metadata#Customers/$entity\",\"Id\":1,\"Name\":\"Customer 1\",\"Type\":\"Retail\"}}," +
+                "{\"id\":\"2\"," +
+                "\"status\":0," +
+                "\"headers\":{\"odata-version\":\"4.0\",\"content-type\":\"application/json\"}, " +
+                "\"body\" :{\"@odata.context\":\"http://tempuri.org/$metadata#Orders/$entity\",\"Id\":1,\"CustomerId\":1,\"Amount\":13}}" +
+                "]}",
+                result);
+        }
+
+#if NETCOREAPP3_1_OR_GREATER
+        [Fact]
+        public async Task WriteBatchResponseAsync_WithStreamCopy_UsingODataUtf8JsonWriter()
+        {
+            Action<IContainerBuilder> configureServices = (builder) =>
+            {
+                builder.AddService<IStreamBasedJsonWriterFactory>(ServiceLifetime.Singleton, _ =>DefaultStreamBasedJsonWriterFactory.Default);
+            };
+
+            var result = await SetupJsonLightBatchWriterAndRunTestAsync(
+                async (jsonLightBatchWriter) =>
+                {
+                    await jsonLightBatchWriter.WriteStartBatchAsync();
+
+                    var customerOperationMessage = await jsonLightBatchWriter.CreateOperationResponseMessageAsync("1");
+                    customerOperationMessage.SetHeader("odata-version", "4.0");

```

Comment:
Not sure these `SetHeader` lines are necessary but if you choose to retain them, maybe change `odata-version` to `OData-Version` since that's what's defined in https://github.com/OData/odata.net/blob/056e8fdb4cdc1ae21a872fdbd5847c527a079b3e/src/Microsoft.OData.Core/ODataConstants.cs#L55